waku/waku_api/rest/admin/handlers.nim

```diff
@@ -0,0 +1,111 @@
+when (NimMajor, NimMinor) < (1, 4):
+  {.push raises: [Defect].}
+else:
+  {.push raises: [].}
+
+import
+  std/strformat,
+  std/sequtils,
+  stew/byteutils,
+  chronicles,
+  json_serialization,
+  presto/route,
+  libp2p/[peerinfo, switch]
+
+import
+  ../../../waku_core,
+  ../../../waku_store,
+  ../../../waku_filter,
+  ../../../waku_relay,
+  ../../../waku_node,
+  ../../../node/peer_manager,
+  ../responses,
+  ../serdes,
+  ./types
+
+export types
+
+logScope:
+  topics = "waku node rest admin api"
+
+const ROUTE_ADMIN_V1_PEERS* = "/admin/v1/peers"
+
+func decodeRequestBody[T](contentBody: Option[ContentBody]) : Result[T, RestApiResponse] =
```

Comment:
Is there a reason why we have this function copied in every handler implementation? Sicne it is using generics, it feels like it should be there only once in the code - obviously, feel free to disregard for this particular PR, but might be worth cleaning up the code if there is no explicit reason to have multiple copies of the func kept around

Reply:
I have the same feelings about. I plan some reorg in a separate PR soon regarding rest api types and related functions. I would keep it as is for now till that small enhancement PR.